jdaviz/app.py

```diff
@@ -612,10 +612,12 @@ def _get_all_subregions(mask, spec_axis_data):
                     this_type = type(value.subset_state)
 
                 # Skip spatial or spectral subsets if only the other is wanted
-                if subset_type == "spectral" and this_type == RoiSubsetState:
-                    continue
-                elif subset_type == "spatial" and this_type == RangeSubsetState:
-                    continue
+                if subset_type == "spectral" or viewer_reference == "spectrum-viewer":
```

Comment:
is hardcoding the viewer-reference string ever going to cause issues (also in line 618)?  (See #1038, for example).  As far as I can tell, we always currently use "spectrum-viewer" for spectrum viewers.  We could check if `viewer` is a `SpecvizProfileView` instance, but I'm not sure if that's better or not.

Reply:
As long as @havok2063 does not overwrite this on MAST side, we should be 99% safe...

Reply:
We currently don't override anything within the `viewer-area` section of the configurations.  I don't think we have any plans to but one never knows.  Items in `settings`, `toolbar`, and `tray` we do reconfigure.

Reply:
Checking for `SpecvizProfileView` seems like it's probably the more stable solution for now.

Reply:
Actually, better to look for `BqplotProfileView`. I'll make that change shortly.